src/Tokenizer/Tokens.php

```diff
@@ -593,39 +594,50 @@ public function getPrevNonWhitespace($index, $whitespaces = null)
      * Get index for closest previous token of given kind.
      * This method is shorthand for getTokenOfKindSibling method.
      *
-     * @param int   $index         token index
-     * @param array $tokens        possible tokens
-     * @param bool  $caseSensitive perform a case sensitive comparison
+     * @param int      $index         token index
+     * @param array    $tokens        possible tokens
+     * @param bool     $caseSensitive perform a case sensitive comparison
+     * @param int|null $limit         false no limit
      *
      * @return int|null
      */
-    public function getPrevTokenOfKind($index, array $tokens = array(), $caseSensitive = true)
+    public function getPrevTokenOfKind($index, array $tokens = array(), $caseSensitive = true, $limit = null)
     {
-        return $this->getTokenOfKindSibling($index, -1, $tokens, $caseSensitive);
+        return $this->getTokenOfKindSibling($index, -1, $limit, $tokens, $caseSensitive);
     }
 
     /**
      * Get index for closest sibling token of given kind.
      *
-     * @param int   $index         token index
-     * @param int   $direction     direction for looking, +1 or -1
-     * @param array $tokens        possible tokens
-     * @param bool  $caseSensitive perform a case sensitive comparison
+     * @param int      $index         token index
+     * @param int      $direction     direction for looking, +1 or -1
+     * @param int|null $limit         null no limit
+     * @param array    $tokens        possible tokens
+     * @param bool     $caseSensitive perform a case sensitive comparison
      *
      * @return int|null
      */
-    public function getTokenOfKindSibling($index, $direction, array $tokens = array(), $caseSensitive = true)
+    public function getTokenOfKindSibling($index, $direction, $limit = null, array $tokens = array(), $caseSensitive = true)
```

Comment:
why do we need it in the first place? what was wrong with checking it on the calling place?
it's not common in our code to check that limit as far as I can see